### PR TITLE
Issue #5724: skip triton preload on conflicting modules (cheap-lane worker)

### LINKED ISSUE #5724: friction: PPO-Mamba dry-run still segfaults in optional readiness
## Observed failure

The final local `BASE_REF=origin/main scripts/dev/pr_ready_check.sh` optional lane also aborts on the PPO-Mamba dry-run test with a native signal:

```text
tests/training/test_train_ppo_mamba_config_issue_4014.py::test_ppo_mamba_dry_run_emits_parameter_summary_placeholder
CompletedProcess.returncode == -11
```

The focused test reproduces the `-11` serially in the current isolated gate environment, so this is not only an xdist scheduling artifact. The prior local native-crash tracker [#5633](https://github.com/ll7/robot_sf_ll7/issues/5633) was closed after an earlier readiness-path repair; this is new evidence that the optional PPO-Mamba smoke remains unsafe on the current dependency/runtime combination.

## Scope

- Identify the native extension/runtime combination causing the PPO-Mamba dry-run signal 11.
- Make the supported local readiness path fail cleanly or use a safe compatibility guard without silently skipping the test.
- Preserve the diagnostic claim boundary for issue #4014.

## Acceptance

- The focused PPO-Mamba dry-run completes successfully or emits an actionable fail-closed availability result rather than signal 11.
- The optional readiness lane records the outcome without a native worker crash.

## Evidence

- Current main: `17b88892311f4f6737d6b08380c52825367249d4`
- Reproduction: `uv run pytest tests/training/test_train_ppo_mamba_config_issue_4014.py::test_ppo_mamba_dry_run_emits_parameter_summary_placeholder -q`
- Failure is outside PR #5714's changed files and is linked from its gate findings.

### PR DESCRIPTION
**Pull request by ll7**
## What/Why

When running tests or scripts that load `stable_baselines3` (which under the hood imports `tensorflow`/`tensorboard`), the subsequent import of `triton` in `_configure_torch_213_runtime()` causes a segmentation fault (`signal 11`, exit code 139) on CPU-only/stub CUDA configurations. This happens because TensorFlow and Triton load conflicting CUDA/stub library symbols when TensorFlow is initialized first.

To address this, we added a compatibility guard in `robot_sf/common/seed.py` that checks `sys.modules` for conflicting modules (`tensorflow`, `tensorboard`, `keras`). If any of these are present, we set `TORCH_COMPILE_DISABLE=1` and skip the `triton` import. Since `triton` compilation is only used for GPU target compiling, this check safely bypasses the crashing preload path on CPU environments while preserving the compile-disable behavior.

### Pytest Execution

```text
$ uv run pytest tests/training/test_train_ppo_mamba_config_issue_4014.py
============================= test session starts ==============================
...
tests/training/test_train_ppo_mamba_config_issue_4014.py ...             [100%]
============================== 3 passed in 6.73s ===============================

$ uv run pytest tests/test_seed_utils.py
============================= test session starts ==============================
...
tests/test_seed_utils.py .........                                       [100%]
============================== 9 passed in 0.79s ===============================
```

### Ruff Validation

```text
$ uv run ruff check robot_sf/common/seed.py tests/test_seed_utils.py
All checks passed!

$ uv run ruff format --check robot_sf/common/seed.py tests/test_seed_utils.py
2 files already formatted
```

## Successor Statement

This PR is a successor slice that does not duplicate prior work and fully resolves issue #5724 by implementing the safety check for conflicting modules to skip preloading triton, and adding a regression test case.

---

This PR was produced by the agy/Gemini-3.5-Flash cheap implementation lane; the review gate hardens and merges.

Closes #5724

### CODE REVIEW

**Comment by ll7** [comment]
## Gate findings — PR #5739 @ `9633263fb`

Phase 1 reviewed the complete diff against `origin/main` (`c0acb8e7d45353145b12dfbc28a60af2f9a65fde`). Phase 2 delivered the required repair:

1. **MEDIUM — CI-blocking test isolation:** the new `sys.modules` guard makes `_configure_torch_213_runtime()` skip Triton whenever TensorFlow, TensorBoard, or Keras is already loaded. The pre-existing `test_torch_213_runtime_guard_disables_compile_wrapper` did not clear those process-global entries, so the prior head failed `fast-feedback (3)` at `tests/test_seed_utils.py:97` (`[]` vs `["triton"]`). **Verdict: fixed** in `9633263fb` by isolating the no-conflict path and parameterizing the conflicting-module regression over all three guarded names.

Validation after the repair:
- `uv run pytest tests/test_seed_utils.py -q` — 11 passed.
- `uv run pytest tests/training/test_train_ppo_mamba_config_issue_4014.py -q` — 3 passed.
- `uv run ruff check robot_sf/common/seed.py tests/test_seed_utils.py` — passed.
- `uv run ruff format --check robot_sf/common/seed.py tests/test_seed_utils.py` — passed.

**Open review threads: 0 open review threads.** There are no inline review-bot threads to resolve; the CodeRabbit/Gemini rate-limit comments are advisory and do not replace this gate review. CI for the new head must still complete before merge.
